components/three_days_left_trial_modal/three_days_left_trial_modal.tsx

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useMemo} from 'react';
+import {useSelector, useDispatch} from 'react-redux';
+import {useIntl} from 'react-intl';
+
+import {isModalOpen} from 'selectors/views/modals';
+import {GlobalState} from 'types/store';
+
+import {closeModal} from 'actions/views/modals';
+
+import {DispatchFunc} from 'mattermost-redux/types/actions';
+
+import {ConsolePages, ModalIdentifiers} from 'utils/constants';
+
+import GenericModal from 'components/generic_modal';
+import GuestAccessSvg from 'components/common/svg_images_components/guest_access_svg';
+import MonitorImacLikeSVG from 'components/common/svg_images_components/monitor_imaclike_svg';
+import SystemRolesSVG from 'components/admin_console/feature_discovery/features/images/system_roles_svg';
+import useOpenPricingModal from 'components/common/hooks/useOpenPricingModal';
+import WorkspaceLimitsPanel from 'components/cloud_usage_modal/workspace_limits_panel';
+import useGetUsage from 'components/common/hooks/useGetUsage';
+import useGetLimits from 'components/common/hooks/useGetLimits';
+
+import ThreeDaysLeftTrialCard, {ThreeDaysLeftTrialCardProps} from './three_days_left_trial_modal_card';
+
+import './three_days_left_trial_modal.scss';
+
+type Props = {
+    onExited?: () => void;
+    limitsOVerpassed: boolean;
+}
+
+function ThreeDaysLeftTrialModal(props: Props): JSX.Element | null {
+    const dispatch = useDispatch<DispatchFunc>();
+    const {formatMessage} = useIntl();
+    const openPricingModal = useOpenPricingModal();
+    const show = useSelector((state: GlobalState) => isModalOpen(state, ModalIdentifiers.THREE_DAYS_LEFT_TRIAL_MODAL));
+    const usage = useGetUsage();
+    const [limits] = useGetLimits();
+
+    // move this to the show three days left so it is easier to test
+    const handleOnClose = async () => {
+        if (props.onExited) {
+            props.onExited();
+        }
+        await dispatch(closeModal(ModalIdentifiers.THREE_DAYS_LEFT_TRIAL_MODAL));
```

Comment:
Does this need to wait?

Reply:
I rather to keep it this way to make sure one modal is closed before the other is shown